src/main/java/org/opensearch/knn/index/MethodComponent.java

```diff
@@ -296,11 +298,24 @@ public static Map<String, Object> getParameterMapWithDefaultsAdded(
     ) {
         Map<String, Object> parametersWithDefaultsMap = new HashMap<>();
         Map<String, Object> userProvidedParametersMap = methodComponentContext.getParameters();
+        Version indexCreationVersion = methodComponentContext.getIndexVersion();
         for (Parameter<?> parameter : methodComponent.getParameters().values()) {
             if (methodComponentContext.getParameters().containsKey(parameter.getName())) {
                 parametersWithDefaultsMap.put(parameter.getName(), userProvidedParametersMap.get(parameter.getName()));
             } else {
-                parametersWithDefaultsMap.put(parameter.getName(), parameter.getDefaultValue());
+                // Picking the right values for the parameters whose values are different based on different index
+                // created version.
+                if (parameter.getName().equals(KNNConstants.METHOD_PARAMETER_EF_SEARCH)) {
+                    parametersWithDefaultsMap.put(parameter.getName(), IndexHyperParametersUtil.getHNSWEFSearchValue(indexCreationVersion));
+                } else if (parameter.getName().equals(KNNConstants.METHOD_PARAMETER_EF_CONSTRUCTION)) {
+                    parametersWithDefaultsMap.put(
+                        parameter.getName(),
+                        IndexHyperParametersUtil.getHNSWEFConstructionValue(indexCreationVersion)
+                    );
+                } else {
+                    parametersWithDefaultsMap.put(parameter.getName(), parameter.getDefaultValue());
+                }
+
             }
```

Comment:
nit: Can this for loop refactored using streams to make it more readable?

Reply:
why we want to do it via streams? I don't see any reason to make this as stream.